LLNL/spack
# Add missing dependencies for Flake8

This PR includes the following changes:

1. Add packages for all of flake8's dependencies
 - was previously downloading from PyPi
2. Rename `py-pep8` to `py-pycodestyle`
 - pep8 module was renamed to pycodestyle [at Guido's request](https://github.com/PyCQA/pycodestyle/issues/466)
3. Add latest version of `py-flake8`
4. Switch flake8 download to GitHub 
 - PyPi URL no longer extrapolates
5. Switch deptype for setuptools to nolink, add checks

The last change is the most important here. `py-flake8` (as well as `py-pyflakes` and `py-pycodestyle`) actually depend on `py-setuptools` at runtime. Otherwise, you get error messages like:
```
Traceback (most recent call last):
  File: "/usr/bin/flake8", line 5, in <module>
    from pkg_resources import load_entry_point
ImportError: No module named pkg_resources
```
Now that the deptype is correct, you can't activate `py-flake8` without also activating `py-setuptools`. Unfortunately, this doesn't help people who don't activate their Python packages. I added checks to `check_dependencies` to make sure setuptools is available before running `run-flake8-tests`. Let me know if you can think of a better way to do this.
